alt-tab-macos/logic/Extensions.swift

```diff
@@ -39,3 +35,25 @@ extension String {
         self.init(string)
     }
 }
+
+// add recursive lookup in subviews for specific type
+extension NSView {
+    func findNestedViews<T: NSView>(subclassOf: T.Type) -> [T] {
+        return recursiveSubviews.compactMap { $0 as? T }
+    }
+
+    var recursiveSubviews: [NSView] {
+        return subviews + subviews.flatMap { $0.recursiveSubviews }
+    }
+}
+
+// add convenience to NSError
+extension NSError {
+    class func make(domain: String, message: String, code: Int = 9999) -> NSError {
+        return NSError(
+                domain: domain,
+                code: code,
+                userInfo: [NSLocalizedDescriptionKey: message, NSLocalizedFailureReasonErrorKey: message]
+        )
+    }
+}
```

Comment:
The motivation here was: I required something that is an NSError or sub-classes it for the error sheet in PreferencesPanel. Why I used NSError was because the Cocoa framework is using that for its exception during the json save (disk/permissions issue) and places very useful informations that describe the error at hand and can be used to be displayed to the user. The PreferencesPanel is using that `.localizedDescription` as a text.

The earlier String extension sub-classed Error but NSError does not sub-class Error (that surprised my a little bit, I expected NSError to sub-class Error to be honest).

Re-using NSError was one simple and fast way to throw application errors, of course without going into an app-wide error/logging overhaul (out of scope this pull request).

Reply:
Oh I see, interesting distinction between `NSError` and `Error`. These frameworks sure have accreted..